consensus/polybft/consensus_runtime.go

```diff
@@ -222,26 +239,35 @@ func (c *consensusRuntime) OnBlockInserted(block *types.Block) {
 		c.logger.Error("on block inserted error", "err", err)
 	}
 
+	var (
+		proposerSnapshot = c.proposerSnapshot.Copy()
```

Comment:
Move this line above this one, because otherwise we would copy and assign proposerSnapshot even when error occurs in epoch-ending blocks but it is not required to have it in that use case.
`if err := c.updateProposerSnapshot(proposerSnapshot, block.Number()); err != nil {`